app/login.go

```diff
@@ -238,6 +238,40 @@ func (a *App) DoLogin(c *request.Context, w http.ResponseWriter, r *http.Request
 	return nil
 }
 
+func (a *App) AttachCloudSessionCookie(c *request.Context, w http.ResponseWriter, r *http.Request) {
+	secure := false
+	if GetProtocol(r) == "https" {
+		secure = true
+	}
+
+	maxAge := *a.Config().ServiceSettings.SessionLengthWebInDays * 60 * 60 * 24
+	domain := a.GetCookieDomain()
+	subpath, _ := utils.GetSubpathFromConfig(a.Config())
+	expiresAt := time.Unix(model.GetMillis()/1000+int64(maxAge), 0)
+
+	var val string
+	if strings.Contains(domain, "localhost") {
```

Comment:
Should we check for cloud here, and only drop the cookie in the event it is cloud? I don't think it's a big deal not to have it, but some more security-conscious on-prem users may appreciate it

Reply:
The cloud check is already there but in the api layer